File: cherrychik/src/main/java/br/com/nixtor/service/PedidoService.java
package br.com.nixtor.service;

import java.util.Date;
import java.util.List;

import br.com.nixtor.model.Pedido;
import br.com.nixtor.model.PedidoRelatorio;

public interface PedidoService {
	
	Pedido salvar(Pedido pedido);
	List<Pedido> listarPedidos();
	Pedido buscarPorId(Long id);
	List<PedidoRelatorio> relatorio();
	List<Pedido> listarPedidosPorData(Date data);
//	double calcularTotal(List<Item> itens);
//	double calcularSubTotal(Item item);
}

File: cherrychik/src/main/webapp/common/services/pedidoService.js
(function(){
	angular
		.module("common.services")
		.factory("pedidoService", ["$http", pedidoService]);
	
	function pedidoService($http) {
		
		var url = "api/pedido/";				
		
		function salvarPedido(pedido) {
			return $http.post(url, pedido).then(function(response){return response.data;});
		}
		
		function listarPedidos() {
			return $http.get(url).then(function(response){return response.data;});
		}
		
		function listarPedidosPorData(data) {
			return $http.post(url + "filtro/", data).then(function(response){return response.data;});
		}
		
		function buscarPorId(id) {
			return $http.get(url + id).then(function(response){return response.data;});
		}
		
		function calcularSubtotal(item) {
			return item.preco * item.quantidade;
		}
		
		function calcularTotal(itens) {
			var total = 0;
			
			for(var i = 0; i < itens.length; i++) {
				total += calcularSubtotal(itens[i]);
			}			
			return total;
		};
		
		function relatorio() {
			return $http.get(url + "relatorio/").then(function(response){return response.data;});
		}
		
		return {			
			salvarPedido: salvarPedido,		
			listarPedidos: listarPedidos,
			listarPedidosPorData: listarPedidosPorData,
			buscarPorId: buscarPorId,
			calcularSubtotal: calcularSubtotal,
			calcularTotal: calcularTotal,
			relatorio: relatorio,			
		}
		
	}
}());